python/cuml/cuml/accel/estimator_proxy.py

```diff
@@ -201,23 +203,22 @@ class ProxyEstimator(class_b):
 
         """
 
-        def __init__(self, *args, **kwargs):
-            self._cpu_model_class = (
-                original_class_a  # Store a reference to the original class
-            )
+        _cpu_model_class = original_class_a
+        _cpu_hyperparams = list(original_class_sig.parameters.keys())
 
+        def __init__(self, *args, **kwargs):
+            # The cuml signature may not align with the sklearn signature.
+            # Additionally, some sklearn models support positional arguments.
+            # To work around this, we
+            # - Bind arguments to the sklearn signature
+            # - Convert the arguments to named parameters
+            # - Translate them to cuml equivalents
+            # - Then forward them on to the cuml class
+            bound = original_class_sig.bind_partial(*args, **kwargs)
             translated_kwargs, self._gpuaccel = self._hyperparam_translator(
-                **kwargs
+                **bound.arguments
             )
-            super().__init__(*args, **translated_kwargs)
-
-            self._cpu_hyperparams = list(
-                inspect.signature(
-                    self._cpu_model_class.__init__
-                ).parameters.keys()
-            )
-
-            self.import_cpu_model()
```

Comment:
I think I know why we don't need this line anymore, the CPU class has already been imported when we get here. But that was already true before this PR, so I am wondering why it was here in the first place and/or if there is a foot gun that we avoid by keeping it? What was your thinking when you deleted it?

Reply:
It was always superfluous and a no-op - `import_cpu_model` bails out immediately if `_cpu_model_class` is already set, and since it was before (and still is) we don't need this call.